### PR TITLE
Removed TryFromRaw, implemented DomainType where needed

### PR DESCRIPTION
**Pull request by greg-szabo**
Closes: #249 

## Description
TryFromRaw was removed and replaced with the DomainType trait. Tests were updated and I found that the ConnectionID integration test fails because the simd application protobuf is outdated. (Commented out for now.)

______

For contributor use:

- [X] Unit tests written
- [ ] Added test to CI if applicable 
- [ ] Updated CHANGELOG_PENDING.md
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Review by romac — APPROVED** [review]
Great stuff, thanks :)

File: relayer-cli/tests/integration.rs [hunk 1]
```diff
@@ -47,14 +47,18 @@ fn simd_chain() -> CosmosSDKChain {
 #[test]
 #[ignore]
 fn query_connection_id() {
+    /* the current informaldev/simd image has an incompatible (old?) protobuf implementation
```
Comment (romac) [inline relayer-cli/tests/integration.rs:50]: What’s the path forward for re-enabling this test? Do we just need to wait for the image to be updated to the new protobuf implementation? What does that mean actually? New protobuf message descriptors?

File: modules/src/ics03_connection/connection.rs [hunk 2]
```diff
@@ -36,6 +37,23 @@ impl TryFromRaw for ConnectionEnd {
...
+            client_id: value.client_id.to_string(),
+            versions: value.versions,
+            state: value.state as i32,
+            counterparty: Some(ibc_proto::ibc::connection::Counterparty {
+                client_id: value.counterparty.client_id.as_str().to_string(),
+                connection_id: value.counterparty.connection_id.to_string(),
+                prefix: Some(ibc_proto::ibc::commitment::MerklePrefix {
+                    key_prefix: value.counterparty.prefix.0,
+                }),
+            }),
```
Comment (adizere) [inline modules/src/ics03_connection/connection.rs:52]:
```suggestion
            counterparty: Some(RawCounterparty::from(value.counterparty)),
```

**Review by adizere — APPROVED** [review]
Cool cool! Thanks, Greg.

File: relayer-cli/tests/integration.rs (hunk 1 above)
Comment (greg-szabo) [reply relayer-cli/tests/integration.rs:50]: Yes, we need a new docker image with an updated simd application. The protobuf message was updated in the SDK and an extra field added. We have more recent protobuf structs than the simd. (The simd is from the times of game-of-zones phase 1 when a bunch of the protobuf looked different.)

**Pull request merged**